src/vm/wren_core.c

```diff
@@ -472,6 +482,11 @@ DEF_PRIMITIVE(fiber_yield1)
 
 DEF_PRIMITIVE(fn_new)
 {
+  RETURN_ERROR("Fn metaclass does not implement 'new()'.");
+}
```

Comment:
Actually now that I look at the code a bit I think I see the disconnect. It's the compiler that decides whether to create a default constructor but when the source for the core code is parsed the [Fn class](https://github.com/munificent/wren/blob/eda3f89266276878ea4c17ca46d635b5c3d43938/src/vm/wren_core.c#L17) is empty and does not yet define its constructor. From the compiler's perspective it sees a class without any constructors and creates a default constructor for it anyways.